bot.js

```diff
@@ -7,504 +7,440 @@ require('dotenv').config();
-				concord.todaysPlayers.forEach(function(player) {
-					var hr = Math.floor((now - player.lastTime) / msHour);
-					var min = Math.floor(((now - player.lastTime) % msHour) / msMinute);
-					if (hr < 1) {
-						numPlayers++;
-						str += player.name + ' ' + min + 'm, ';
-					}
-				});
-				str = str.slice(0, -2)
-				if (str.length > 0) str += ') ';
-				if (numPlayers == 0 && concord.todaysPlayers.length > 0) {
-					var hr = Math.floor((now - concord.todaysPlayers[0].lastTime) / msHour);
-					channel.setTopic(`No one's here! :eyes: The last players were seen over ${hr} hour(s) ago. [${nowString}]`)
-						.then(updated => console.log(`round1-concord: ${updated.topic}`))
-						.catch(console.error);
-				} else {
-					channel.setTopic(`${numPlayers} player(s) in the last hour. :eyes: <:TFTI:483651827984760842> ${str}[${nowString}]`)
-						.then(updated => console.log(`round1-concord: ${updated.topic}`))
-						.catch(console.error);
-				}
-			} else if (channel.name === 'round1-sanjose') {
-				sanJose.todaysPlayers.forEach(function(player) {
-					var hr = Math.floor((now - player.lastTime) / msHour);
-					var min = Math.floor(((now - player.lastTime) % msHour) / msMinute);
-					if (hr < 1) {
-						numPlayers++;
-						str += player.name + ' ' + min + 'm, ';
-					}
-				});
-				str = str.slice(0, -2)
-				if (str.length > 0) str += ') ';
-				if (numPlayers == 0 && sanJose.todaysPlayers.length > 0) {
-					var hr = Math.floor((now - sanJose.todaysPlayers[0].lastTime) / msHour);
-					channel.setTopic(`No one's here! :eyes: The last players were seen over ${hr} hour(s) ago. [${nowString}]`)
-						.then(updated => console.log(`round1-sanjose: ${updated.topic}`))
-						.catch(console.error);
-				} else {
-					channel.setTopic(`${numPlayers} player(s) in the last hour. :eyes: <:TFTI:483651827984760842> ${str}[${nowString}]`)
-						.then(updated => console.log(`round1-sanjose: ${updated.topic}`))
-						.catch(console.error);
-				}
-			}
-		});
-		setTimeout(function() {
-			updateChannelTopics()
-		}, 60000);
-	}
+function updateChannelTopic(loc, channel) {
+  const currentTime = new Date();
+  const nowString = timeString(currentTime, loc.timeZone);
+
+  let numPlayers = 0;
+  const playerNamesTimes = [];
+
+  loc.todaysPlayers.forEach(function(player) {
+    const timeSinceSeen = timeDifferential(currentTime, player.lastTime);
+    if (timeSinceSeen.h < 1) {
+      numPlayers++;
+      playerNamesTimes.push(`${player.name} ${timeSinceSeen.m}m`);
+    }
+  });
+
+  let topic;
+  if (loc.todaysPlayers.length === 0) {
+    // TODO: Perhaps include the day's start time (local time per arcade)
+    topic = `${nowString}: 0 players today.`;
+  } else if (numPlayers === 0) {
+    const timeSinceSeen = timeDifferential(currentTime, loc.todaysPlayers[0].lastTime);
+    topic = `${nowString}: Everyone(${loc.todaysPlayers.length}) left! :eyes: Last player seen: ${loc.todaysPlayers[0].name} ${timeSinceSeen.str} ago.`;
```

Comment:
Prefer space between `Everyone` and the total number of today's players.
```
topic = `${nowString}: Everyone (${loc.todaysPlayers.length}) left!...`;
```

Reply:
Also kind of strange to put it after `Everyone`. It's not immediately clear what the number means.

Reply:
It was ambiguous on purpose, I just wanted to drop it in to see if it was useful. You're right, though. How about something like "`All 8 players left!`"?

Reply:
It kind of sounds like there were 8 players together at one point... maybe `All 8 players today left!`